tests/core/model_management/test_local_job_predictor.py

```diff
@@ -0,0 +1,304 @@
+
+
+def test_purge_retention_time(predictor_type_cfg):
+    """Test that purging old models works as expected"""
+    predictor = local_predictor(retention_duration="1d10s", **predictor_type_cfg)
+    model_future = predictor.predict(
+        SampleModule(),
+        "run",
+        sample_input=SampleInputType(),
+    )
+    model_future.wait()
+    retrieved_future = predictor.get_job_future(model_future.id)
+    assert retrieved_future is model_future
+    model_future._completion_time = model_future._completion_time - timedelta(days=2)
+    with pytest.raises(CaikitCoreException):
+        predictor.get_job_future(model_future.id)
+    assert not Path(retrieved_future.save_path).exists()
+    assert not Path(retrieved_future.save_path).parent.exists()
+
+
+@pytest.mark.parametrize(
+    "test_params",
+    [
+        ("1d", timedelta(days=1)),
+        ("0.01s", timedelta(seconds=0.01)),
+        ("1d12h3m0.2s", timedelta(days=1, hours=12, minutes=3, seconds=0.2)),
+    ],
+)
+def test_retention_duration_parse(test_params):
+    """Make sure the regex for the duration can parse all expected durations"""
+    trainer = local_predictor(retention_duration=test_params[0])
+    assert trainer._retention_duration == test_params[1]
+
+
+def test_get_into_return_error(predictor_type_cfg):
+    """Test that failed prediction returns error properly"""
+    predictor = local_predictor(**predictor_type_cfg)
+
+    model_future = predictor.predict(
+        SampleModule(),
+        "run",
+        sample_input=SampleInputType("hello"),
+        throw=True,
+        error="Error",
+    )
+    # assert model_future.get_info().status == TrainingStatus.RUNNING
+    # assert not model_future.get_info().status.is_terminal
+
+    # Let the training proceed and wait for it to complete
+    model_future.wait()
+    assert model_future.get_info().status == PredictionJobStatus.ERRORED
+    assert model_future.get_info().status.is_terminal
+    assert isinstance(model_future.get_info().errors, list)
+    assert isinstance(model_future.get_info().errors[0], RuntimeError)
+    assert str(model_future.get_info().errors[0]) == "Error"
+
+
+def test_retry_duplicate_external_id():
+    """Test that a prediction can be retried safely reusing an external ID"""
+    predictor = local_predictor()
+    predictor_id = "my-prediction"
+
+    # First try should fail
+    with pytest.raises(CaikitCoreException):
+        model_future = predictor.predict(
+            SampleModule(),
+            "run",
+            external_prediction_id=predictor_id,
+            sample_input=SampleInputType("hello"),
+            throw=True,
+            error="Error",
+        )
+        model_future.result()
+
+    # Second time should succeed
+    model_future = predictor.predict(
+        SampleModule(),
+        "run",
+        external_prediction_id=predictor_id,
+        sample_input=SampleInputType("hello"),
+    )
+    assert model_future.result()
```

Comment:
Since the `result` of the prediction job is a bit more important content-wise than the training job, I think we also need at least one test that makes sure the `result()` is the correct proto object (that might exist in the server tests below that I haven't gotten to yet)